edweiss-app/app/(app)/lectures/remotecontrol/index.tsx

```diff
@@ -61,159 +49,44 @@ const RemoteControlScreen: ApplicationRoute = () => {
-            if (wasRecording) setIsRecording(true);
-
-            // On sucess update hook
-            setCurrentPage(currentPage + 1);
-
-        }
-        Vibration.vibrate(100);
-    }
-
-    function handleMic() {
-        if (isRecording) {
-            stopRecoding();
-            setIsRecording(false);
-        } else {
-            startRecording();
-            setIsRecording(true);
-        }
-
-        Vibration.vibrate(100);
-        setIsRecording(!isRecording);
-    }
-
-    // Portrait display for the screen
-    const setPortrait = async () => {
-        await ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.PORTRAIT_UP);
-    };
-
     return (
         <>
+            {<AbstractRmtCrl
+                handleRight={() => handleRight(isRecording, currentPage, totalPages, setIsRecording, setCurrentPage, stopRecording)}
+                handleLeft={() => handleLeft(isRecording, currentPage, setIsRecording, setCurrentPage, stopRecording)}
+                handleMic={() => handleMic(isRecording, setIsRecording, startRecording, stopRecording)}
+                isRecording={isRecording}
+            />}
+        </>
+    );
+};
 
-            <RouteHeader disabled title={"Lecture's Slides"} />
-
-
-            <TView mt={100} mb={100} justifyContent='center' alignItems='center'>
-                <TText size={25} mb={'md'}> {t(`showtime:showtime_title`)}</TText>
-                <TText size={35} >{t(`showtime:rmt_cntl_title`)}</TText>
-                {isRecording ? (
-                    <TText mt={15} size={15} color='red'> {t(`showtime:recording_start`)} </TText>
-                ) : (
-                    <TText mt={15} size={15} color='green'> {t(`showtime:tap_to_start_recording`)} </TText>
-                )}
-            </TView>
+export default RemoteControlScreen;
 
-            <TView mr={20} ml={20} flexDirection='row' justifyContent='space-between'>
-                <TTouchableOpacity backgroundColor='blue' borderColor='surface0' b={1} p={20} pt={60} pb={60} radius={'lg'} onPress={handleLeft}>
-                    <Icon size={70} name='chevron-back-outline' color='base'></Icon>
-                </TTouchableOpacity>
-                <TTouchableOpacity backgroundColor='blue' borderColor='surface0' b={1} p={20} pt={60} pb={60} radius={'lg'} onPress={handleRight}>
-                    <Icon size={70} name='chevron-forward-outline' color='base'></Icon>
-                </TTouchableOpacity>
-            </TView>
 
 
-            <TView mt={70} mb={30} justifyContent='center' alignItems='center'>
-                <TTouchableOpacity backgroundColor={isRecording ? 'red' : 'base'} borderColor={isRecording ? 'base' : 'red'} b={5} p={10} radius={1000} onPress={handleMic}>
-                    <Icon size={70} name='mic-outline' color={isRecording ? 'base' : 'red'}></Icon>
-                </TTouchableOpacity>
-            </TView>
+export const startRecording = async () => {
```

Comment:
Are you using these functions outside of RemoteControlScreen? Because if not I do not know if it is a good practice to put these as export functions available to many files if it is only used internally. Is it for keeping it more organized? I would prefer just maintaining like it was before inside the application route function

Reply:
That's a great point of yours. Indeed, it not a good pratice to do this 🤔. However, I wanted to separate the `utilsFunctions` from the `remotecontrol/index` and the `abstractrmtctl`. This way we could reuse the audio 
 processing functions in another context 🚀. Won't that be cool that you could generate to-dos, directly using your `EdI` (Educational Intelligence), and monitor your `assignment` vocally 🤟 ? Do you see my point, or would you like a small refactoring ?

Reply:
No its ok if you see a line in the future where this structure is advantageous then all good